hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/erasurecode/ErasureCodingWorker.java

```diff
@@ -172,4 +175,19 @@ public void shutDown() {
   public float getXmitWeight() {
     return xmitWeight;
   }
+
+  public void setStripedReconstructionPoolSize(int size) {
+    Preconditions.checkArgument(size > 0,
+        DFS_DN_EC_RECONSTRUCTION_THREADS_KEY + " should be greater than 0");
+    this.stripedReconstructionPool.setCorePoolSize(size);
+    this.stripedReconstructionPool.setMaximumPoolSize(size);
+  }
+
+  @VisibleForTesting
+  public int getStripedReconstructionPoolSize() {
+    int poolSize = this.stripedReconstructionPool.getCorePoolSize();
+    Preconditions.checkArgument(poolSize == this.stripedReconstructionPool.getMaximumPoolSize(),
+        "The maximum pool size should be equal to core pool size");
```

Comment:
I think this isn't required, this looks like validating the ``ThreadPoolExecutor``, that ain't a hadoop thing & isn't required here in the scope either